src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java

```diff
@@ -56,10 +57,78 @@ public void findAssociatedNotLinkedFilesSuccess() throws Exception {
     }
 
     @Test
-    public void findAssociatedNotLinkedFilesForEmptySearchDir() throws Exception {
+    public void testFindAssociatedNotLinkedFilesForEmptySearchDir() throws Exception {
         when(filePreferences.shouldStoreFilesRelativeToBibFile()).thenReturn(false);
         AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(databaseContext, filePreferences, autoLinkPrefs);
         List<LinkedFile> actual = util.findAssociatedNotLinkedFiles(entry);
         assertEquals(Collections.emptyList(), actual);
     }
+
+        /* @BeforeEach
+    public void setUp(@TempDir Path folder) throws Exception {
+        A = folder.resolve("A");
+        Files.createDirectory(A);
+        B = folder.resolve("B");
+        Files.createDirectory(B);
+        path = A.resolve("CiteKey.pdf");
+        Files.createFile(path);
+        entry.setCitationKey("CiteKey");
+        when(filePreferences.getExternalFileTypes())
+                .thenReturn(FXCollections.observableSet(new TreeSet<>(ExternalFileTypes.getDefaultExternalFileTypes())));
+    } */
+
+    @Test
+    public void testFileLinksAfterMoving() throws Exception {
+        // Run "Automatically set file links" - check that the bib file was not modified
+        path = path.getParent();
+
+        Path A = path.resolve("A");
+        Files.createDirectory(A);
+        Path B = path.resolve("B");
+        Files.createDirectory(B);
+        Path filePath = A.resolve("Test.pdf");
+        Files.createFile(filePath);
+        entry.setCitationKey("Test");
+
+        LinkedFile linkedFile = new LinkedFile("desc", filePath, "PDF");
+        List<LinkedFile> listLinked = new ArrayList<>();
+        listLinked.add(linkedFile);
+        entry.setFiles(listLinked);
+
+        Path destination = B.resolve("Test.pdf");
+        Files.move(A.resolve("Test.pdf"), destination, StandardCopyOption.REPLACE_EXISTING);
+
+        AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(databaseContext, filePreferences, autoLinkPrefs);
+        AutoSetFileLinksUtil.RelinkedResults results = util.relinkingFiles(entry.getFiles());
+        List<LinkedFile> list = results.relinkedFiles();
+        List<String> exceptions = results.exceptions();
+        assertEquals(Collections.emptyList(), exceptions);
+
+        // Change Entry to match required result and run method on bib
+        LinkedFile linkedDestFile = new LinkedFile("desc", B.resolve("Test.pdf"), "PDF");
+        List<LinkedFile> listLinked2 = new ArrayList<>();
```

Comment:
Having numbers as variable name suffix is bad style. Please craft self-descriptive variable names. E.g., "existingFiles" or something.